src/core/src/CoreMain.py

```diff
@@ -115,6 +125,16 @@ def __init__(self, argv):
             composite_logger.log_debug("Completed exception handling.\n")
 
         finally:
+            # clean up temp folder after all operation execution is finished from Core
+            # Note to reviewer: This only deletes all *.list files currently
+            if bootstrapper.env_layer is not None \
+                    and execution_config is not None \
+                    and execution_config.temp_folder is not None \
+                    and os.path.exists(execution_config.temp_folder):
+                composite_logger.log_debug("Deleting all files of certain format from temp folder [FileFormat={0}][TempFolderLocation={1}]"
+                                           .format(Constants.ARTIFACT_IDENTIFIER_FOR_ITEMS_TO_CLEANUP_UNDER_TEMP_FOLDER, str(execution_config.temp_folder)))
+                bootstrapper.env_layer.file_system.delete_artifacts_from_dir(execution_config.temp_folder, Constants.ARTIFACT_IDENTIFIER_FOR_ITEMS_TO_CLEANUP_UNDER_TEMP_FOLDER)
```

Comment:
Encapsulate reused code into method, and invoke wherever necessary.

Reply:
Resolved